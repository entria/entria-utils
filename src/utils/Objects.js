// @flow
export const concat = (objectOne: Object, objectTwo: Object): Object => {
  const concatenated = {
    ...objectOne,
  };

  Object.keys(objectTwo).forEach(key => {
    if (typeof concatenated[key] === 'object') {
      concatenated[key] = {
        ...concatenated[key],
        ...objectTwo[key],
      };
    } else {
      concatenated[key] = objectTwo[key];
    }
  });

  return concatenated;
};

export const clear = (dirtyObject: Object): Object => {
  const hasValue = (value: any): boolean => {
    if (
      value === undefined ||
      value === null ||
      value.length === 0 ||
      (typeof value === typeof {} && Object.keys(value).length === 0)
    ) {
      return false;
    }

    return true;
  };

  return Object.keys(dirtyObject).filter(key => hasValue(dirtyObject[key])).reduce(
    (acc, key) => ({
      ...acc,
      [key]: dirtyObject[key],
    }),
    {},
  );
};
