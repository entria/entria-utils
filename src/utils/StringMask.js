// @flow

import Mask from 'string-mask';

export default function(value: ?string, format: ?string): string {
  return new Mask(format).apply(value);
}
