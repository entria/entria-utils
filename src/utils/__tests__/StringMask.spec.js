import StringMask from '../StringMask';

import { setupTest } from '../../../test/utils';

beforeEach(async () => setupTest());

it('should format string for CPF mask', () => {
  expect(StringMask('99999999999', '000.000.000-00')).toMatchSnapshot();
});

it('should only format 6 characters of the CPF mask', () => {
  expect(StringMask('999999', '000.000.000-00')).toMatchSnapshot();
});

it('should format string for CEP mask', () => {
  expect(StringMask('99999999', '00000-000')).toMatchSnapshot();
});

it('should format string for CNPJ mask', () => {
  expect(StringMask('99999999999999', '00.000.000/0000-00')).toMatchSnapshot();
});
