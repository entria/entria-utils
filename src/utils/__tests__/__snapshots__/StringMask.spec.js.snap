// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format string for CEP mask 1`] = `"99999-999"`;

exports[`should format string for CNPJ mask 1`] = `"99.999.999/9999-99"`;

exports[`should format string for CPF mask 1`] = `"999.999.999-99"`;

exports[`should only format 6 characters of the CPF mask 1`] = `"999.999."`;
